CMakeLists.txt

```diff
@@ -38,31 +38,48 @@ endif()
 # Set available build types for CMake GUIs.
 # A different build type can still be set by -DCMAKE_BUILD_TYPE=...
 set_property(CACHE CMAKE_BUILD_TYPE PROPERTY
-  STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+  STRINGS "Debug" "Dev" "Release" "MinSizeRel" "RelWithDebInfo")
 
 # Set default build type.
 if(NOT CMAKE_BUILD_TYPE)
-  message(STATUS "CMAKE_BUILD_TYPE not given; setting to 'RelWithDebInfo'.")
-  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build." FORCE)
+  message(STATUS "CMAKE_BUILD_TYPE not given; setting to 'Dev'.")
+  set(CMAKE_BUILD_TYPE "Dev" CACHE STRING "Choose the type of build." FORCE)
 endif()
 
 # Version tokens
-include(GetGitRevisionDescription)
-file(TO_NATIVE_PATH ${CMAKE_CURRENT_LIST_DIR}/.git GIT_DIR)
-get_git_head_revision(GIT_REFSPEC NVIM_VERSION_COMMIT)
-if(NOT NVIM_VERSION_COMMIT)
-  set(NVIM_VERSION_COMMIT "?")
-endif()
+# On release:
+#  1. Set NVIM_VERSION_PRERELEASE to "". Tag (annotated tag: git tag -a) this commit.
+#  2. Set NVIM_VERSION_PRERELEASE to "-alpha". Increase version and commit.
 set(NVIM_VERSION_MAJOR 0)
 set(NVIM_VERSION_MINOR 0)
 set(NVIM_VERSION_PATCH 0)
 set(NVIM_VERSION_PRERELEASE "-alpha")
-# TODO(justinmk): UTC time would be nice here #1071
-git_timestamp(GIT_TIMESTAMP)
-# TODO(justinmk): do not set this for "release" builds #1071
-if(GIT_TIMESTAMP)
-  set(NVIM_VERSION_BUILD "+${GIT_TIMESTAMP}")
+
+include(GetGitRevisionDescription)
+file(TO_NATIVE_PATH ${CMAKE_CURRENT_LIST_DIR}/.git GIT_DIR)
+if(NVIM_VERSION_PRERELEASE)
+  get_git_head_revision(GIT_REFSPEC NVIM_VERSION_COMMIT)
+  if(NOT NVIM_VERSION_COMMIT)
+    set(NVIM_VERSION_COMMIT "?")
```

Comment:
I think we decided to leave this out. If commit not found, that line shouldn't appear in `:version`.